### ISSUE TITLE
Inconsistency in implementation from paper

### ISSUE THREAD

**Issue by krishnakanthnakka** [peiwang062/swag#1]
In Eq. 11 from the paper, the softmax is operated over the entire feature map i.e., across all channels (m) and spatial locations(n) in the denominator term.

However, from the code, torch.nn.softmax2D() operation does the softmax across channels for every spatial location. It does not smooth the entire feature map.

Can you please clarify this? Thanks in advance if I missed something.

**Comment by peiwang062** [comment]
As the paper explains, there are many ways to smooth the activations. This is just one of possible strategies.
In our experiments, we found that as long as the mapping can smooth the activations and increase the entropy, it should work as well.
We have updated the codes to clarify this. thank you.
